File: output/gemma4-e4b/avatar/Python/Java/atcoder_ABC178_B.java
import java.util.Scanner;
import java.lang.Math;

public class atcoder_ABC178_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read the four integers a, b, c, d from standard input.
        // Using long to prevent potential overflow during multiplication.
        if (!scanner.hasNextLong()) {
            // Handle case where input might be missing or malformed
            return;
        }
        
        long a = scanner.nextLong();
        long b = scanner.nextLong();
        long c = scanner.nextLong();
        long d = scanner.nextLong();
        
        scanner.close();
        
        // Calculate the four products
        long prod1 = a * c;
        long prod2 = b * d;
        long prod3 = a * d;
        long prod4 = b * c;
        
        // Find the maximum of the four products
        long result = Math.max(prod1, Math.max(prod2, Math.max(prod3, prod4)));
        
        System.out.println(result);
    }
}

File: output/gemma4-e4b/avatar/Java/Python/atcoder_ABC178_B.py
import sys

def solve():
    # Read all inputs from stdin and convert them to integers
    try:
        data = sys.stdin.read().split()
        if not data:
            return
        
        # We only need the first 4 elements
        a = [int(x) for x in data[:4]]
    except Exception:
        return

    # Calculate the four products and find the maximum
    # max(a[0]*a[2], a[1]*a[3], a[0]*a[3], a[1]*a[2])
    result = max(a[0] * a[2], a[1] * a[3], a[0] * a[3], a[1] * a[2])
    
    print(result)

if __name__ == "__main__":
    solve()